### PR TITLE
Postgresql 15 support

### PR DESCRIPTION
**Pull request by pcnc**
## What kind of change does this PR introduce?

* Upgrades PostgreSQL to v15
* Includes a PostgREST major version bump, generating a breaking change when it comes to multicolumn indexes
* Also brings extension updates in tow
* Bumps Packer builder size from `t4g.2xlarge` to `c6g.4xlarge` for extra compute capacity when compiling `plv8`, which does not finish compilation within defined timeout
   * turns out builds now finish in ~40min instead of ~1h
* Docker image WIP; currently experiencing build timeouts due to increased plv8 build times 

### Additional details
* `pgx` updated to 0.6.0-alpha
  * ~[ARM support PR pending approval](https://github.com/tcdi/pgx/pull/870)~ merged
* `pg_graphql` and `pg_jsonschema`
  * ~currently built off specific pg15_support branches~ merged
* `timescaledb`
  * pinned to specific commit in `main` branch, will update once officially released
* `pljava`
  * [PR pending approval](https://github.com/tada/pljava/pull/413)
  * temporarily building off our [own fork](https://github.com/supabase/pljava)
* `plpgsql_check` ~is disabled due to erroring out on Postgres start~
  * `[182007] FATAL:  requested tranche is not registered`
  * ~[PR pending approval](https://github.com/okbob/plpgsql_check/pull/131)~ merged
  * pinned to specific commit in `master` branch, will update once release is tagged

### CODE REVIEW

File: .github/workflows/ami-release.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - develop
+      - pcnc/pg15
```
Comment (pcnc) [inline .github/workflows/ami-release.yml]: This needs removal pre-merge

File: ansible/vars.yml [hunk 2]
```diff
@@ -47,31 +47,32 @@ adminapi_release_checksum:
...
-plpgsql_check_release_checksum: sha1:3f3ca49e38ee87392fd62f1a62a481c445e1c6c2
+plpgsql_check_release: "2.2.3"
+plpgsql_check_release_checksum: sha256:bb531dcf55366e21c88544289d617a9e923d122e8d919bb37205e562bfea2e9b
 
 pg_safeupdate_release: "1.4"
 pg_safeupdate_release_checksum: sha1:942dacd0ebce6123944212ffb3d6b5a0c09174f9
 
-timescaledb_release: "2.6.1"
+timescaledb_release: 121631c70fdf08c285ac9201d0767ed99e4891c8
+# timescaledb_release: "2.8.1" //TODO: update version after pg15 support is rolled out: https://github.com/timescale/timescaledb/pulls?q=is%3Apr+label%3Apg15+is%3Aclosed
```
Comment (pcnc) [inline ansible/vars.yml:72]: TimescaleDB version needs update once PG15 support is officially released

File: ansible/vars.yml [hunk 3]
```diff
@@ -100,13 +101,15 @@ libsodium_release_checksum: sha1:795b73e3f92a362fabee238a71735579bf46bb97
...
-pg_graphql_release: "v0.5.0"
+#TODO: update versions when both stable pgx and pg_graphql versions are released
+pg_graphql_pgx_version: "0.6.0-alpha.1"
+pg_graphql_release: "pg15_support"
 
-pg_jsonschema_pgx_version: "0.5.6"
-pg_jsonschema_release: "v0.1.1"
+#TODO: update versions when both stable pgx and pg_jsonschema versions are released
+pg_jsonschema_pgx_version: "0.6.0-alpha.1"
+pg_jsonschema_release: "pg15_support"
```
Comment (pcnc) [inline ansible/vars.yml]: Needs update to point to proper pg_graphql and pg_jsonschema releases once PG15 support is merged

File: ansible/tasks/postgres-extensions/10-timescaledb.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 - name: timescaledb - bootstrap
   shell:
-    cmd: "./bootstrap -DAPACHE_ONLY=1"
+    cmd: "./bootstrap -DAPACHE_ONLY=1 -DEXPERIMENTAL=ON" #TODO: remove this after official support is rolled out: https://github.com/timescale/timescaledb/pulls?q=is%3Apr+label%3Apg15+is%3Aclosed
```
Comment (pcnc) [inline ansible/tasks/postgres-extensions/10-timescaledb.yml:18]: Remove this once official TimescaleDB support is rolled out

File: ansible/files/postgresql_config/postgresql.conf.j2 [hunk 5]
```diff
@@ -688,7 +688,8 @@ default_text_search_config = 'pg_catalog.english'
 #local_preload_libraries = ''
 #session_preload_libraries = ''
 
-shared_preload_libraries = 'pg_stat_statements, pg_stat_monitor, pgaudit, plpgsql, plpgsql_check, pg_cron, pg_net, pgsodium, timescaledb, auto_explain'	# (change requires restart)
+#TODO: add plpgsql_check back to the list after "FATAL:  requested tranche is not registered" is fixed
+shared_preload_libraries = 'pg_stat_statements, pg_stat_monitor, pgaudit, plpgsql, pg_cron, pg_net, pgsodium, timescaledb, auto_explain'	# (change requires restart)
```
Comment (pcnc) [inline ansible/files/postgresql_config/postgresql.conf.j2]: plpgsql_check needs to be added back once fixed

File: ansible/tasks/postgres-extensions/12-pljava.yml [hunk 6]
```diff
@@ -9,19 +9,27 @@
...
-  unarchive:
-    remote_src: yes
-    src: /tmp/pljava-{{ pljava_release }}.tar.gz
-    dest: /tmp
+- name: pljava - download latest release
   become: yes
+  git:
+    repo: https://github.com/supabase/pljava.git
+    dest: /tmp/pljava-{{ pljava_release }}
+    version: "master"
```
Comment (pcnc) [inline ansible/tasks/postgres-extensions/12-pljava.yml]: Revert to using proper pljava releases once pg15 support gets merged into main repo

File: ansible/tasks/postgres-extensions/08-plpgsql_check.yml [hunk 7]
```diff
@@ -6,19 +6,27 @@
...
+#     dest: /tmp/plpgsql_check-{{ plpgsql_check_release }}.tar.gz
+#     checksum: "{{ plpgsql_check_release_checksum }}"
+#     timeout: 60
+
+# - name: plpgsql_check - unpack archive
+#   unarchive:
+#     remote_src: yes
+#     src: /tmp/plpgsql_check-{{ plpgsql_check_release }}.tar.gz
+#     dest: /tmp
+#   become: yes
```
Comment (pcnc) [inline ansible/tasks/postgres-extensions/08-plpgsql_check.yml]: Revert to using proper plpgsql_check releases once pg15 fix gets merged into main repo

File: ansible/tasks/postgres-extensions/12-pljava.yml (hunk 6 above)
Comment (darora) [reply ansible/tasks/postgres-extensions/12-pljava.yml]: (ideally a specific commit sha, though I guess if we're controlling the branch it's not the end of the world)

File: ansible/tasks/postgres-extensions/13-plv8.yml [hunk 8]
```diff
@@ -19,6 +19,7 @@
       - libc++abi1
       - ninja-build
       - python
+      - clang
```
Comment (darora) [inline ansible/tasks/postgres-extensions/13-plv8.yml]: we have gcc being used for postgres etc - should we switch it out for clang? or just move forward with both for now?

**Review by darora — APPROVED** [review]

File: ansible/tasks/postgres-extensions/12-pljava.yml (hunk 6 above)
Comment (pcnc) [reply ansible/tasks/postgres-extensions/12-pljava.yml]: Very fair point - will address this together with the merge conflicts

File: ansible/tasks/postgres-extensions/13-plv8.yml (hunk 8 above)
Comment (pcnc) [reply ansible/tasks/postgres-extensions/13-plv8.yml]: Good point. I think we can remove it. Makefiles call gcc everywhere anyway, the clang dependency was just picked off v3.0.0 docs but I have the feeling those haven't been updated for v3.1.4

**Pull request merged**